# swagger-client
API integration layer for linked biological objects.   __Source:__ https://github.com/biolink/biolink-api/

This Python package is automatically generated by the [Swagger Codegen](https://github.com/swagger-api/swagger-codegen) project:

- API version: 1.1.14
- Package version: 1.0.0
- Build package: io.swagger.codegen.v3.generators.python.PythonClientCodegen

## Requirements.

Python 2.7 and 3.4+

## Installation & Usage
### pip install

If the python package is hosted on Github, you can install directly from Github

```sh
pip install git+https://github.com/ManuJazz/biolink-python-client.git
```
(you may need to run `pip` with root permission: `sudo pip install git+https://github.com/ManuJazz/biolink-python-client.git`)

Then import the package:
```python
import swagger_client 
```

### Setuptools

Install via [Setuptools](http://pypi.python.org/pypi/setuptools).

```sh
python setup.py install --user
```
(or `sudo python setup.py install` to install the package for all users)

Then import the package:
```python
import swagger_client
```

## Getting Started

Please follow the [installation procedure](#installation--usage) and then run the following:

```python
from __future__ import print_function
import time
import swagger_client
from swagger_client.rest import ApiException
from pprint import pprint

# create an instance of the API class
api_instance = swagger_client.AssociationApi(swagger_client.ApiClient(configuration))
association_type = 'association_type_example' # str | Association type, eg gene_phenotype
rows = 100 # int | number of rows (optional) (default to 100)
start = 56 # int | beginning row (optional)
evidence = 'evidence_example' # str | Object id, e.g. ECO:0000501 (for IEA; Includes inferred by default) or a specific publication or other supporting object, e.g. ZFIN:ZDB-PUB-060503-2 (optional)
unselect_evidence = false # bool | If true, excludes evidence objects in response (optional) (default to false)
exclude_automatic_assertions = false # bool | If true, excludes associations that involve IEAs (ECO:0000501) (optional) (default to false)
use_compact_associations = false # bool | If true, returns results in compact associations format (optional) (default to false)
subject = 'subject_example' # str | Subject CURIE (optional)
object = 'object_example' # str | Object CURIE (optional)

try:
    # Returns list of matching associations of a given type
    api_response = api_instance.get_association_by_subject_and_assoc_type(association_type, rows=rows, start=start, evidence=evidence, unselect_evidence=unselect_evidence, exclude_automatic_assertions=exclude_automatic_assertions, use_compact_associations=use_compact_associations, subject=subject, object=object)
    pprint(api_response)
except ApiException as e:
    print("Exception when calling AssociationApi->get_association_by_subject_and_assoc_type: %s\n" % e)

# create an instance of the API class
api_instance = swagger_client.AssociationApi(swagger_client.ApiClient(configuration))
object_category = 'object_category_example' # str | Category of entity at link Object (target), e.g. gene, disease, phenotype
subject_category = 'subject_category_example' # str | Category of entity at link Subject (source), e.g. gene, disease, phenotype
rows = 100 # int | number of rows (optional) (default to 100)
start = 56 # int | beginning row (optional)
evidence = 'evidence_example' # str | Object id, e.g. ECO:0000501 (for IEA; Includes inferred by default) or a specific publication or other supporting object, e.g. ZFIN:ZDB-PUB-060503-2 (optional)
unselect_evidence = false # bool | If true, excludes evidence objects in response (optional) (default to false)
exclude_automatic_assertions = false # bool | If true, excludes associations that involve IEAs (ECO:0000501) (optional) (default to false)
use_compact_associations = false # bool | If true, returns results in compact associations format (optional) (default to false)
subject = 'subject_example' # str | Subject CURIE (optional)
object = 'object_example' # str | Object CURIE (optional)
subject_taxon = 'subject_taxon_example' # str | Subject taxon ID, e.g. NCBITaxon:9606 (Includes inferred associations, by default) (optional)
object_taxon = 'object_taxon_example' # str | Object taxon ID, e.g. NCBITaxon:10090 (Includes inferred associations, by default) (optional)
relation = 'relation_example' # str | Filter by relation CURIE, e.g. RO:0002200 (has_phenotype), RO:0002607 (is marker for), RO:HOM0000017 (orthologous to), etc. (optional)

try:
    # Returns list of matching associations between a given subject and object category
    api_response = api_instance.get_association_by_subject_and_object_category_search(object_category, subject_category, rows=rows, start=start, evidence=evidence, unselect_evidence=unselect_evidence, exclude_automatic_assertions=exclude_automatic_assertions, use_compact_associations=use_compact_associations, subject=subject, object=object, subject_taxon=subject_taxon, object_taxon=object_taxon, relation=relation)
    pprint(api_response)
except ApiException as e:
    print("Exception when calling AssociationApi->get_association_by_subject_and_object_category_search: %s\n" % e)

# create an instance of the API class
api_instance = swagger_client.AssociationApi(swagger_client.ApiClient(configuration))
subject_category = 'subject_category_example' # str | Category of entity at link Subject (source), e.g. gene, disease, phenotype
rows = 100 # int | number of rows (optional) (default to 100)
start = 56 # int | beginning row (optional)
evidence = 'evidence_example' # str | Object id, e.g. ECO:0000501 (for IEA; Includes inferred by default) or a specific publication or other supporting object, e.g. ZFIN:ZDB-PUB-060503-2 (optional)
unselect_evidence = false # bool | If true, excludes evidence objects in response (optional) (default to false)
exclude_automatic_assertions = false # bool | If true, excludes associations that involve IEAs (ECO:0000501) (optional) (default to false)
use_compact_associations = false # bool | If true, returns results in compact associations format (optional) (default to false)
subject_taxon = 'subject_taxon_example' # str | Subject taxon ID, e.g. NCBITaxon:9606 (Includes inferred associations, by default) (optional)
object_taxon = 'object_taxon_example' # str | Object taxon ID, e.g. NCBITaxon:10090 (Includes inferred associations, by default) (optional)
relation = 'relation_example' # str | Filter by relation CURIE, e.g. RO:0002200 (has_phenotype), RO:0002607 (is marker for), RO:HOM0000017 (orthologous to), etc. (optional)

try:
    # Returns list of matching associations for a given subject category
    api_response = api_instance.get_association_by_subject_category_search(subject_category, rows=rows, start=start, evidence=evidence, unselect_evidence=unselect_evidence, exclude_automatic_assertions=exclude_automatic_assertions, use_compact_associations=use_compact_associations, subject_taxon=subject_taxon, object_taxon=object_taxon, relation=relation)
    pprint(api_response)
except ApiException as e:
    print("Exception when calling AssociationApi->get_association_by_subject_category_search: %s\n" % e)

# create an instance of the API class
api_instance = swagger_client.AssociationApi(swagger_client.ApiClient(configuration))
id = 'id_example' # str | identifier for an association, e.g. f5ba436c-f851-41b3-9d9d-bb2b5fc879d4

try:
    # Returns the association with a given identifier
    api_response = api_instance.get_association_object(id)
    pprint(api_response)
except ApiException as e:
    print("Exception when calling AssociationApi->get_association_object: %s\n" % e)

# create an instance of the API class
api_instance = swagger_client.AssociationApi(swagger_client.ApiClient(configuration))
object = 'object_example' # str | Return associations pointing to this node, e.g. MP:0013765. Can also be a biological entity such as a gene
subject = 'subject_example' # str | Return associations emanating from this node, e.g. MGI:1342287 (If ID is from an ontology then results would include inferred associations, by default)
rows = 100 # int | number of rows (optional) (default to 100)
start = 56 # int | beginning row (optional)
evidence = 'evidence_example' # str | Object id, e.g. ECO:0000501 (for IEA; Includes inferred by default) or a specific publication or other supporting object, e.g. ZFIN:ZDB-PUB-060503-2 (optional)
unselect_evidence = false # bool | If true, excludes evidence objects in response (optional) (default to false)
exclude_automatic_assertions = false # bool | If true, excludes associations that involve IEAs (ECO:0000501) (optional) (default to false)
use_compact_associations = false # bool | If true, returns results in compact associations format (optional) (default to false)

try:
    # Returns associations connecting two entities
    api_response = api_instance.get_associations_between(object, subject, rows=rows, start=start, evidence=evidence, unselect_evidence=unselect_evidence, exclude_automatic_assertions=exclude_automatic_assertions, use_compact_associations=use_compact_associations)
    pprint(api_response)
except ApiException as e:
    print("Exception when calling AssociationApi->get_associations_between: %s\n" % e)

# create an instance of the API class
api_instance = swagger_client.AssociationApi(swagger_client.ApiClient(configuration))
subject = 'subject_example' # str | Return associations emanating from this node, e.g. NCBIGene:84570, ZFIN:ZDB-GENE-050417-357 (If ID is from an ontology then results would include inferred associations, by default)
rows = 100 # int | number of rows (optional) (default to 100)
start = 56 # int | beginning row (optional)
evidence = 'evidence_example' # str | Object id, e.g. ECO:0000501 (for IEA; Includes inferred by default) or a specific publication or other supporting object, e.g. ZFIN:ZDB-PUB-060503-2 (optional)
unselect_evidence = false # bool | If true, excludes evidence objects in response (optional) (default to false)
exclude_automatic_assertions = false # bool | If true, excludes associations that involve IEAs (ECO:0000501) (optional) (default to false)
use_compact_associations = false # bool | If true, returns results in compact associations format (optional) (default to false)
object_taxon = 'object_taxon_example' # str | Object taxon ID, e.g. NCBITaxon:10090 (Includes inferred associations, by default) (optional)
relation = 'relation_example' # str | Filter by relation CURIE, e.g. RO:0002200 (has_phenotype), RO:0002607 (is marker for), RO:HOM0000017 (orthologous to), etc. (optional)

try:
    # Returns list of matching associations starting from a given subject (source)
    api_response = api_instance.get_associations_from(subject, rows=rows, start=start, evidence=evidence, unselect_evidence=unselect_evidence, exclude_automatic_assertions=exclude_automatic_assertions, use_compact_associations=use_compact_associations, object_taxon=object_taxon, relation=relation)
    pprint(api_response)
except ApiException as e:
    print("Exception when calling AssociationApi->get_associations_from: %s\n" % e)

# create an instance of the API class
api_instance = swagger_client.AssociationApi(swagger_client.ApiClient(configuration))
object = 'object_example' # str | Return associations pointing to this node, e.g. specifying MP:0013765 will return all genes, variants, strains, etc. annotated with this term. Can also be a biological entity such as a gene
rows = 100 # int | number of rows (optional) (default to 100)
start = 56 # int | beginning row (optional)
evidence = 'evidence_example' # str | Object id, e.g. ECO:0000501 (for IEA; Includes inferred by default) or a specific publication or other supporting object, e.g. ZFIN:ZDB-PUB-060503-2 (optional)
unselect_evidence = false # bool | If true, excludes evidence objects in response (optional) (default to false)
exclude_automatic_assertions = false # bool | If true, excludes associations that involve IEAs (ECO:0000501) (optional) (default to false)
use_compact_associations = false # bool | If true, returns results in compact associations format (optional) (default to false)

try:
    # Returns list of matching associations pointing to a given object (target)
    api_response = api_instance.get_associations_to(object, rows=rows, start=start, evidence=evidence, unselect_evidence=unselect_evidence, exclude_automatic_assertions=exclude_automatic_assertions, use_compact_associations=use_compact_associations)
    pprint(api_response)
except ApiException as e:
    print("Exception when calling AssociationApi->get_associations_to: %s\n" % e)
```

## Documentation for API Endpoints

All URIs are relative to */api*

Class | Method | HTTP request | Description
------------ | ------------- | ------------- | -------------
*AssociationApi* | [**get_association_by_subject_and_assoc_type**](docs/AssociationApi.md#get_association_by_subject_and_assoc_type) | **GET** /association/type/{association_type} | Returns list of matching associations of a given type
*AssociationApi* | [**get_association_by_subject_and_object_category_search**](docs/AssociationApi.md#get_association_by_subject_and_object_category_search) | **GET** /association/find/{subject_category}/{object_category} | Returns list of matching associations between a given subject and object category
*AssociationApi* | [**get_association_by_subject_category_search**](docs/AssociationApi.md#get_association_by_subject_category_search) | **GET** /association/find/{subject_category} | Returns list of matching associations for a given subject category
*AssociationApi* | [**get_association_object**](docs/AssociationApi.md#get_association_object) | **GET** /association/{id} | Returns the association with a given identifier
*AssociationApi* | [**get_associations_between**](docs/AssociationApi.md#get_associations_between) | **GET** /association/between/{subject}/{object} | Returns associations connecting two entities
*AssociationApi* | [**get_associations_from**](docs/AssociationApi.md#get_associations_from) | **GET** /association/from/{subject} | Returns list of matching associations starting from a given subject (source)
*AssociationApi* | [**get_associations_to**](docs/AssociationApi.md#get_associations_to) | **GET** /association/to/{object} | Returns list of matching associations pointing to a given object (target)
*BioentityApi* | [**get_anatomy_gene_associations**](docs/BioentityApi.md#get_anatomy_gene_associations) | **GET** /bioentity/anatomy/{id}/genes | Returns genes associated with a given anatomy
*BioentityApi* | [**get_anatomy_gene_by_taxon_associations**](docs/BioentityApi.md#get_anatomy_gene_by_taxon_associations) | **GET** /bioentity/anatomy/{id}/genes/{taxid} | Returns gene IDs for all genes associated with a given anatomy, filtered by taxon
*BioentityApi* | [**get_case_disease_associations**](docs/BioentityApi.md#get_case_disease_associations) | **GET** /bioentity/case/{id}/diseases | Returns diseases associated with a case
*BioentityApi* | [**get_case_genotype_associations**](docs/BioentityApi.md#get_case_genotype_associations) | **GET** /bioentity/case/{id}/genotypes | Returns genotypes associated with a case
*BioentityApi* | [**get_case_model_associations**](docs/BioentityApi.md#get_case_model_associations) | **GET** /bioentity/case/{id}/models | Returns models associated with a case
*BioentityApi* | [**get_case_phenotype_associations**](docs/BioentityApi.md#get_case_phenotype_associations) | **GET** /bioentity/case/{id}/phenotypes | Returns phenotypes associated with a case
*BioentityApi* | [**get_case_variant_associations**](docs/BioentityApi.md#get_case_variant_associations) | **GET** /bioentity/case/{id}/variants | Returns variants associated with a case
*BioentityApi* | [**get_disease_case_associations**](docs/BioentityApi.md#get_disease_case_associations) | **GET** /bioentity/disease/{id}/cases | Returns cases associated with a disease
*BioentityApi* | [**get_disease_gene_associations**](docs/BioentityApi.md#get_disease_gene_associations) | **GET** /bioentity/disease/{id}/genes | Returns genes associated with a disease
*BioentityApi* | [**get_disease_genotype_associations**](docs/BioentityApi.md#get_disease_genotype_associations) | **GET** /bioentity/disease/{id}/genotypes | Returns genotypes associated with a disease
*BioentityApi* | [**get_disease_model_associations**](docs/BioentityApi.md#get_disease_model_associations) | **GET** /bioentity/disease/{id}/models | Returns associations to models of the disease
*BioentityApi* | [**get_disease_model_taxon_associations**](docs/BioentityApi.md#get_disease_model_taxon_associations) | **GET** /bioentity/disease/{id}/models/{taxon} | Returns associations to models of the disease constrained by taxon
*BioentityApi* | [**get_disease_pathway_associations**](docs/BioentityApi.md#get_disease_pathway_associations) | **GET** /bioentity/disease/{id}/pathways | Returns pathways associated with a disease
*BioentityApi* | [**get_disease_phenotype_associations**](docs/BioentityApi.md#get_disease_phenotype_associations) | **GET** /bioentity/disease/{id}/phenotypes | Returns phenotypes associated with disease
*BioentityApi* | [**get_disease_publication_associations**](docs/BioentityApi.md#get_disease_publication_associations) | **GET** /bioentity/disease/{id}/publications | Returns publications associated with a disease
*BioentityApi* | [**get_disease_substance_associations**](docs/BioentityApi.md#get_disease_substance_associations) | **GET** /bioentity/disease/{id}/treatment | Returns substances associated with a disease
*BioentityApi* | [**get_disease_variant_associations**](docs/BioentityApi.md#get_disease_variant_associations) | **GET** /bioentity/disease/{id}/variants | Returns variants associated with a disease
*BioentityApi* | [**get_function_associations**](docs/BioentityApi.md#get_function_associations) | **GET** /bioentity/function/{id} | Returns annotations associated to a function term
*BioentityApi* | [**get_function_gene_associations**](docs/BioentityApi.md#get_function_gene_associations) | **GET** /bioentity/function/{id}/genes | Returns genes associated to a GO term
*BioentityApi* | [**get_function_publication_associations**](docs/BioentityApi.md#get_function_publication_associations) | **GET** /bioentity/function/{id}/publications | Returns publications associated to a GO term
*BioentityApi* | [**get_function_taxon_associations**](docs/BioentityApi.md#get_function_taxon_associations) | **GET** /bioentity/function/{id}/taxons | Returns taxons associated to a GO term
*BioentityApi* | [**get_gene_anatomy_associations**](docs/BioentityApi.md#get_gene_anatomy_associations) | **GET** /bioentity/gene/{id}/anatomy | Returns anatomical entities associated with a gene
*BioentityApi* | [**get_gene_case_associations**](docs/BioentityApi.md#get_gene_case_associations) | **GET** /bioentity/gene/{id}/cases | Returns cases associated with a gene
*BioentityApi* | [**get_gene_disease_associations**](docs/BioentityApi.md#get_gene_disease_associations) | **GET** /bioentity/gene/{id}/diseases | Returns diseases associated with gene
*BioentityApi* | [**get_gene_expression_associations**](docs/BioentityApi.md#get_gene_expression_associations) | **GET** /bioentity/gene/{id}/expression/anatomy | Returns expression events for a gene
*BioentityApi* | [**get_gene_function_associations**](docs/BioentityApi.md#get_gene_function_associations) | **GET** /bioentity/gene/{id}/function | Returns function associations for a gene
*BioentityApi* | [**get_gene_genotype_associations**](docs/BioentityApi.md#get_gene_genotype_associations) | **GET** /bioentity/gene/{id}/genotypes | Returns genotypes associated with a gene
*BioentityApi* | [**get_gene_homolog_associations**](docs/BioentityApi.md#get_gene_homolog_associations) | **GET** /bioentity/gene/{id}/homologs | Returns homologs for a gene
*BioentityApi* | [**get_gene_interactions**](docs/BioentityApi.md#get_gene_interactions) | **GET** /bioentity/gene/{id}/interactions | Returns interactions for a gene
*BioentityApi* | [**get_gene_model_associations**](docs/BioentityApi.md#get_gene_model_associations) | **GET** /bioentity/gene/{id}/models | Returns models associated with a gene
*BioentityApi* | [**get_gene_ortholog_disease_associations**](docs/BioentityApi.md#get_gene_ortholog_disease_associations) | **GET** /bioentity/gene/{id}/ortholog/diseases | Return diseases associated with orthologs of a gene
*BioentityApi* | [**get_gene_ortholog_phenotype_associations**](docs/BioentityApi.md#get_gene_ortholog_phenotype_associations) | **GET** /bioentity/gene/{id}/ortholog/phenotypes | Return phenotypes associated with orthologs for a gene
*BioentityApi* | [**get_gene_pathway_associations**](docs/BioentityApi.md#get_gene_pathway_associations) | **GET** /bioentity/gene/{id}/pathways | Returns pathways associated with gene
*BioentityApi* | [**get_gene_phenotype_associations**](docs/BioentityApi.md#get_gene_phenotype_associations) | **GET** /bioentity/gene/{id}/phenotypes | Returns phenotypes associated with gene
*BioentityApi* | [**get_gene_publication_associations**](docs/BioentityApi.md#get_gene_publication_associations) | **GET** /bioentity/gene/{id}/publications | Returns publications associated with a gene
*BioentityApi* | [**get_gene_variant_associations**](docs/BioentityApi.md#get_gene_variant_associations) | **GET** /bioentity/gene/{id}/variants | Returns variants associated with a gene
*BioentityApi* | [**get_generic_associations**](docs/BioentityApi.md#get_generic_associations) | **GET** /bioentity/{id}/associations | Returns associations for an entity regardless of the type
*BioentityApi* | [**get_generic_object**](docs/BioentityApi.md#get_generic_object) | **GET** /bioentity/{id} | Returns basic info on object of any type
*BioentityApi* | [**get_generic_object_by_type**](docs/BioentityApi.md#get_generic_object_by_type) | **GET** /bioentity/{type}/{id} | Return basic info on an object for a given type
*BioentityApi* | [**get_genotype_case_associations**](docs/BioentityApi.md#get_genotype_case_associations) | **GET** /bioentity/genotype/{id}/cases | Returns cases associated with a genotype
*BioentityApi* | [**get_genotype_disease_associations**](docs/BioentityApi.md#get_genotype_disease_associations) | **GET** /bioentity/genotype/{id}/diseases | Returns diseases associated with a genotype
*BioentityApi* | [**get_genotype_gene_associations**](docs/BioentityApi.md#get_genotype_gene_associations) | **GET** /bioentity/genotype/{id}/genes | Returns genes associated with a genotype
*BioentityApi* | [**get_genotype_genotype_associations**](docs/BioentityApi.md#get_genotype_genotype_associations) | **GET** /bioentity/genotype/{id}/genotypes | Returns genotypes-genotype associations
*BioentityApi* | [**get_genotype_model_associations**](docs/BioentityApi.md#get_genotype_model_associations) | **GET** /bioentity/genotype/{id}/models | Returns models associated with a genotype
*BioentityApi* | [**get_genotype_phenotype_associations**](docs/BioentityApi.md#get_genotype_phenotype_associations) | **GET** /bioentity/genotype/{id}/phenotypes | Returns phenotypes associated with a genotype
*BioentityApi* | [**get_genotype_publication_associations**](docs/BioentityApi.md#get_genotype_publication_associations) | **GET** /bioentity/genotype/{id}/publications | Returns publications associated with a genotype
*BioentityApi* | [**get_genotype_variant_associations**](docs/BioentityApi.md#get_genotype_variant_associations) | **GET** /bioentity/genotype/{id}/variants | Returns genotypes-variant associations
*BioentityApi* | [**get_goterm_gene_associations**](docs/BioentityApi.md#get_goterm_gene_associations) | **GET** /bioentity/goterm/{id}/genes | Returns associations to GO terms for a gene
*BioentityApi* | [**get_model_case_associations**](docs/BioentityApi.md#get_model_case_associations) | **GET** /bioentity/model/{id}/cases | Returns cases associated with a model
*BioentityApi* | [**get_model_disease_associations**](docs/BioentityApi.md#get_model_disease_associations) | **GET** /bioentity/model/{id}/diseases | Returns diseases associated with a model
*BioentityApi* | [**get_model_gene_associations**](docs/BioentityApi.md#get_model_gene_associations) | **GET** /bioentity/model/{id}/genes | Returns genes associated with a model
*BioentityApi* | [**get_model_genotype_associations**](docs/BioentityApi.md#get_model_genotype_associations) | **GET** /bioentity/model/{id}/genotypes | Returns genotypes associated with a model
*BioentityApi* | [**get_model_phenotype_associations**](docs/BioentityApi.md#get_model_phenotype_associations) | **GET** /bioentity/model/{id}/phenotypes | Returns phenotypes associated with a model
*BioentityApi* | [**get_model_publication_associations**](docs/BioentityApi.md#get_model_publication_associations) | **GET** /bioentity/model/{id}/publications | Returns publications associated with a model
*BioentityApi* | [**get_model_variant_associations**](docs/BioentityApi.md#get_model_variant_associations) | **GET** /bioentity/model/{id}/variants | Returns variants associated with a model
*BioentityApi* | [**get_pathway_disease_associations**](docs/BioentityApi.md#get_pathway_disease_associations) | **GET** /bioentity/pathway/{id}/diseases | Returns diseases associated with a pathway
*BioentityApi* | [**get_pathway_gene_associations**](docs/BioentityApi.md#get_pathway_gene_associations) | **GET** /bioentity/pathway/{id}/genes | Returns genes associated with a pathway
*BioentityApi* | [**get_pathway_phenotype_associations**](docs/BioentityApi.md#get_pathway_phenotype_associations) | **GET** /bioentity/pathway/{id}/phenotypes | Returns phenotypes associated with a pathway
*BioentityApi* | [**get_phenotype_anatomy_associations**](docs/BioentityApi.md#get_phenotype_anatomy_associations) | **GET** /bioentity/phenotype/{id}/anatomy | Returns anatomical entities associated with a phenotype
*BioentityApi* | [**get_phenotype_case_associations**](docs/BioentityApi.md#get_phenotype_case_associations) | **GET** /bioentity/phenotype/{id}/cases | Returns cases associated with a phenotype
*BioentityApi* | [**get_phenotype_disease_associations**](docs/BioentityApi.md#get_phenotype_disease_associations) | **GET** /bioentity/phenotype/{id}/diseases | Returns diseases associated with a phenotype
*BioentityApi* | [**get_phenotype_gene_associations**](docs/BioentityApi.md#get_phenotype_gene_associations) | **GET** /bioentity/phenotype/{id}/genes | Returns genes associated with a phenotype
*BioentityApi* | [**get_phenotype_gene_by_taxon_associations**](docs/BioentityApi.md#get_phenotype_gene_by_taxon_associations) | **GET** /bioentity/phenotype/{id}/gene/{taxid}/ids | Returns gene IDs for all genes associated with a given phenotype, filtered by taxon
*BioentityApi* | [**get_phenotype_genotype_associations**](docs/BioentityApi.md#get_phenotype_genotype_associations) | **GET** /bioentity/phenotype/{id}/genotypes | Returns genotypes associated with a phenotype
*BioentityApi* | [**get_phenotype_pathway_associations**](docs/BioentityApi.md#get_phenotype_pathway_associations) | **GET** /bioentity/phenotype/{id}/pathways | Returns pathways associated with a phenotype
*BioentityApi* | [**get_phenotype_publication_associations**](docs/BioentityApi.md#get_phenotype_publication_associations) | **GET** /bioentity/phenotype/{id}/publications | Returns publications associated with a phenotype
*BioentityApi* | [**get_phenotype_variant_associations**](docs/BioentityApi.md#get_phenotype_variant_associations) | **GET** /bioentity/phenotype/{id}/variants | Returns variants associated with a phenotype
*BioentityApi* | [**get_publication_disease_associations**](docs/BioentityApi.md#get_publication_disease_associations) | **GET** /bioentity/publication/{id}/diseases | Returns diseases associated with a publication
*BioentityApi* | [**get_publication_gene_associations**](docs/BioentityApi.md#get_publication_gene_associations) | **GET** /bioentity/publication/{id}/genes | Returns genes associated with a publication
*BioentityApi* | [**get_publication_genotype_associations**](docs/BioentityApi.md#get_publication_genotype_associations) | **GET** /bioentity/publication/{id}/genotypes | Returns genotypes associated with a publication
*BioentityApi* | [**get_publication_model_associations**](docs/BioentityApi.md#get_publication_model_associations) | **GET** /bioentity/publication/{id}/models | Returns models associated with a publication
*BioentityApi* | [**get_publication_phenotype_associations**](docs/BioentityApi.md#get_publication_phenotype_associations) | **GET** /bioentity/publication/{id}/phenotypes | Returns phenotypes associated with a publication
*BioentityApi* | [**get_publication_variant_associations**](docs/BioentityApi.md#get_publication_variant_associations) | **GET** /bioentity/publication/{id}/variants | Returns variants associated with a publication
*BioentityApi* | [**get_substance_participant_in_associations**](docs/BioentityApi.md#get_substance_participant_in_associations) | **GET** /bioentity/substance/{id}/participant_in | Returns associations between an activity and process and the specified substance
*BioentityApi* | [**get_substance_role_associations**](docs/BioentityApi.md#get_substance_role_associations) | **GET** /bioentity/substance/{id}/roles | Returns associations between given drug and roles
*BioentityApi* | [**get_substance_treats_associations**](docs/BioentityApi.md#get_substance_treats_associations) | **GET** /bioentity/substance/{id}/treats | Returns substances associated with a disease
*BioentityApi* | [**get_variant_case_associations**](docs/BioentityApi.md#get_variant_case_associations) | **GET** /bioentity/variant/{id}/cases | Returns cases associated with a variant
*BioentityApi* | [**get_variant_disease_associations**](docs/BioentityApi.md#get_variant_disease_associations) | **GET** /bioentity/variant/{id}/diseases | Returns diseases associated with a variant
*BioentityApi* | [**get_variant_gene_associations**](docs/BioentityApi.md#get_variant_gene_associations) | **GET** /bioentity/variant/{id}/genes | Returns genes associated with a variant
*BioentityApi* | [**get_variant_genotype_associations**](docs/BioentityApi.md#get_variant_genotype_associations) | **GET** /bioentity/variant/{id}/genotypes | Returns genotypes associated with a variant
*BioentityApi* | [**get_variant_model_associations**](docs/BioentityApi.md#get_variant_model_associations) | **GET** /bioentity/variant/{id}/models | Returns models associated with a variant
*BioentityApi* | [**get_variant_phenotype_associations**](docs/BioentityApi.md#get_variant_phenotype_associations) | **GET** /bioentity/variant/{id}/phenotypes | Returns phenotypes associated with a variant
*BioentityApi* | [**get_variant_publication_associations**](docs/BioentityApi.md#get_variant_publication_associations) | **GET** /bioentity/variant/{id}/publications | Returns publications associated with a variant
*BioentitysetApi* | [**get_entity_set_associations**](docs/BioentitysetApi.md#get_entity_set_associations) | **GET** /bioentityset/associations | Returns compact associations for a given input set
*BioentitysetApi* | [**get_entity_set_graph_resource**](docs/BioentitysetApi.md#get_entity_set_graph_resource) | **GET** /bioentityset/graph | TODO Graph object spanning all entities
*BioentitysetApi* | [**get_entity_set_summary**](docs/BioentitysetApi.md#get_entity_set_summary) | **GET** /bioentityset/descriptor/counts | Summary statistics for objects associated
*BioentitysetApi* | [**get_over_representation**](docs/BioentitysetApi.md#get_over_representation) | **GET** /bioentityset/overrepresentation | Summary statistics for objects associated
*BioentitysethomologsApi* | [**get_entity_set_homologs**](docs/BioentitysethomologsApi.md#get_entity_set_homologs) | **GET** /bioentityset/homologs/ | Returns homology associations for a given input set of genes
*BioentitysetslimmerApi* | [**get_entity_set_anatomy_slimmer**](docs/BioentitysetslimmerApi.md#get_entity_set_anatomy_slimmer) | **GET** /bioentityset/slimmer/anatomy | For a given gene(s), summarize its annotations over a defined set of slim
*BioentitysetslimmerApi* | [**get_entity_set_function_slimmer**](docs/BioentitysetslimmerApi.md#get_entity_set_function_slimmer) | **GET** /bioentityset/slimmer/function | For a given gene(s), summarize its annotations over a defined set of slim
*BioentitysetslimmerApi* | [**get_entity_set_phenotype_slimmer**](docs/BioentitysetslimmerApi.md#get_entity_set_phenotype_slimmer) | **GET** /bioentityset/slimmer/phenotype | For a given gene(s), summarize its annotations over a defined set of slim
*CamApi* | [**get_activity_collection**](docs/CamApi.md#get_activity_collection) | **GET** /cam/activity | Returns list of models
*CamApi* | [**get_instance_object**](docs/CamApi.md#get_instance_object) | **GET** /cam/instance/{id} | Returns list of matches
*CamApi* | [**get_model_collection**](docs/CamApi.md#get_model_collection) | **GET** /cam/model | Returns list of ALL models
*CamApi* | [**get_model_contributors**](docs/CamApi.md#get_model_contributors) | **GET** /cam/model/contributors | Returns list of all contributors across all models
*CamApi* | [**get_model_instances**](docs/CamApi.md#get_model_instances) | **GET** /cam/instances | Returns list of all instances
*CamApi* | [**get_model_object**](docs/CamApi.md#get_model_object) | **GET** /cam/model/{id} | Returns a complete model
*CamApi* | [**get_model_properties**](docs/CamApi.md#get_model_properties) | **GET** /cam/model/properties | Returns list of all properties used across all models
*CamApi* | [**get_model_property_values**](docs/CamApi.md#get_model_property_values) | **GET** /cam/model/property_values | Returns list property-values for all models
*CamApi* | [**get_model_query**](docs/CamApi.md#get_model_query) | **GET** /cam/model/query | Returns list of models matching query
*CamApi* | [**get_physical_interaction**](docs/CamApi.md#get_physical_interaction) | **GET** /cam/physical_interaction | Returns list of models
*EvidencegraphApi* | [**get_evidence_graph_object**](docs/EvidencegraphApi.md#get_evidence_graph_object) | **GET** /evidence/graph/{id} | Returns evidence graph object for a given association
*EvidencegraphApi* | [**get_evidence_graph_table**](docs/EvidencegraphApi.md#get_evidence_graph_table) | **GET** /evidence/graph/{id}/table | Returns evidence as a association_results object given an association
*GenomefeaturesApi* | [**get_features_within_resource**](docs/GenomefeaturesApi.md#get_features_within_resource) | **GET** /genome/features/within/{build}/{reference}/{begin}/{end} | Returns list of matches
*GraphApi* | [**get_edge_resource**](docs/GraphApi.md#get_edge_resource) | **GET** /graph/edges/from/{id} | Returns edges emanating from a given node
*GraphApi* | [**get_node_resource**](docs/GraphApi.md#get_node_resource) | **GET** /graph/node/{id} | Returns a graph node
*IdentifiermapperApi* | [**get_identifier_mapper**](docs/IdentifiermapperApi.md#get_identifier_mapper) | **GET** /identifier/mapper/{source}/{target}/ | TODO maps a list of identifiers from a source to a target
*IdentifierprefixesApi* | [**get_prefix_collection**](docs/IdentifierprefixesApi.md#get_prefix_collection) | **GET** /identifier/prefixes/ | Returns list of prefixes
*IdentifierprefixesApi* | [**get_prefix_contract**](docs/IdentifierprefixesApi.md#get_prefix_contract) | **GET** /identifier/prefixes/contract/{uri} | Returns contracted URI
*IdentifierprefixesApi* | [**get_prefix_expand**](docs/IdentifierprefixesApi.md#get_prefix_expand) | **GET** /identifier/prefixes/expand/{id} | Returns expanded URI
*IndividualApi* | [**get_individual**](docs/IndividualApi.md#get_individual) | **GET** /individual/{id} | Returns list of matches
*IndividualApi* | [**get_pedigree**](docs/IndividualApi.md#get_pedigree) | **GET** /individual/pedigree/{id} | Returns list of matches
*MartApi* | [**get_mart_case_associations_resource**](docs/MartApi.md#get_mart_case_associations_resource) | **GET** /mart/case/{object_category}/{taxon} | Bulk download of case associations
*MartApi* | [**get_mart_disease_associations_resource**](docs/MartApi.md#get_mart_disease_associations_resource) | **GET** /mart/disease/{object_category}/{taxon} | Bulk download of disease associations
*MartApi* | [**get_mart_gene_associations_resource**](docs/MartApi.md#get_mart_gene_associations_resource) | **GET** /mart/gene/{object_category}/{taxon} | Bulk download of gene associations
*MartApi* | [**get_mart_ortholog_associations_resource**](docs/MartApi.md#get_mart_ortholog_associations_resource) | **GET** /mart/ortholog/{taxon1}/{taxon2} | Bulk download of orthologs
*MartApi* | [**get_mart_paralog_associations_resource**](docs/MartApi.md#get_mart_paralog_associations_resource) | **GET** /mart/paralog/{taxon1}/{taxon2} | Bulk download of paralogs
*MetadataApi* | [**get_metadata_for_datasets**](docs/MetadataApi.md#get_metadata_for_datasets) | **GET** /metadata/datasets | Get metadata for all datasets from SciGraph
*MmeApi* | [**post_disease_mme**](docs/MmeApi.md#post_disease_mme) | **POST** /mme/disease | Match a patient to diseases based on their phenotypes
*MmeApi* | [**post_fly_mme**](docs/MmeApi.md#post_fly_mme) | **POST** /mme/fly | Match a patient to fruit fly genes based on similar phenotypes
*MmeApi* | [**post_mouse_mme**](docs/MmeApi.md#post_mouse_mme) | **POST** /mme/mouse | Match a patient to mouse genes based on similar phenotypes
*MmeApi* | [**post_nematode_mme**](docs/MmeApi.md#post_nematode_mme) | **POST** /mme/nematode | Match a patient to nematode genes based on similar phenotypes
*MmeApi* | [**post_zebrafish_mme**](docs/MmeApi.md#post_zebrafish_mme) | **POST** /mme/zebrafish | Match a patient to zebrafish genes based on similar phenotypes
*NlpannotateApi* | [**get_annotate**](docs/NlpannotateApi.md#get_annotate) | **GET** /nlp/annotate/ | Annotate a given text using SciGraph annotator
*NlpannotateApi* | [**get_annotate_entities**](docs/NlpannotateApi.md#get_annotate_entities) | **GET** /nlp/annotate/entities | Annotate a given content using SciGraph annotator and get all entities from content
*NlpannotateApi* | [**post_annotate**](docs/NlpannotateApi.md#post_annotate) | **POST** /nlp/annotate/ | Annotate a given text using SciGraph annotator
*NlpannotateApi* | [**post_annotate_entities**](docs/NlpannotateApi.md#post_annotate_entities) | **POST** /nlp/annotate/entities | Annotate a given content using SciGraph annotator and get all entities from content
*OntolApi* | [**get_extract_ontology_subgraph_resource**](docs/OntolApi.md#get_extract_ontology_subgraph_resource) | **GET** /ontol/subgraph/{ontology}/{node} | Extract a subgraph from an ontology
*OntolApi* | [**get_information_content_resource**](docs/OntolApi.md#get_information_content_resource) | **GET** /ontol/information_content/{subject_category}/{object_category}/{subject_taxon} | Returns information content (IC) for a set of relevant ontology classes
*OntolApi* | [**post_extract_ontology_subgraph_resource**](docs/OntolApi.md#post_extract_ontology_subgraph_resource) | **POST** /ontol/subgraph/{ontology}/{node} | Extract a subgraph from an ontology
*OntolidentifierApi* | [**get_ontol_identifier_resource**](docs/OntolidentifierApi.md#get_ontol_identifier_resource) | **GET** /ontol/identifier/ | Fetches a map from CURIEs/IDs to labels
*OntolidentifierApi* | [**post_ontol_identifier_resource**](docs/OntolidentifierApi.md#post_ontol_identifier_resource) | **POST** /ontol/identifier/ | Fetches a map from CURIEs/IDs to labels
*OntollabelerApi* | [**get_ontol_labeler_resource**](docs/OntollabelerApi.md#get_ontol_labeler_resource) | **GET** /ontol/labeler/ | Fetches a map from CURIEs/IDs to labels
*OntologyApi* | [**get_ontology_subset**](docs/OntologyApi.md#get_ontology_subset) | **GET** /ontology/subset/{id} | Returns meta data of an ontology subset (slim)
*OntologyApi* | [**get_ontology_term**](docs/OntologyApi.md#get_ontology_term) | **GET** /ontology/term/{id} | Returns meta data of an ontology term
*OntologyApi* | [**get_ontology_term_graph**](docs/OntologyApi.md#get_ontology_term_graph) | **GET** /ontology/term/{id}/graph | Returns graph of an ontology term
*OntologyApi* | [**get_ontology_term_subgraph**](docs/OntologyApi.md#get_ontology_term_subgraph) | **GET** /ontology/term/{id}/subgraph | Extract a subgraph from an ontology term
*OntologyApi* | [**get_ontology_term_subsets**](docs/OntologyApi.md#get_ontology_term_subsets) | **GET** /ontology/term/{id}/subsets | Returns subsets (slims) associated to an ontology term
*OntologyApi* | [**get_ontology_terms_shared_ancestor**](docs/OntologyApi.md#get_ontology_terms_shared_ancestor) | **GET** /ontology/shared/{subject}/{object} | Returns the ancestor ontology terms shared by two ontology terms
*OwlontologyApi* | [**get_dl_query**](docs/OwlontologyApi.md#get_dl_query) | **GET** /owl/ontology/dlquery/{query} | Placeholder - use OWLery for now
*OwlontologyApi* | [**get_sparql_query**](docs/OwlontologyApi.md#get_sparql_query) | **GET** /owl/ontology/sparql/{query} | Placeholder - use direct SPARQL endpoint for now
*PairsimApi* | [**get_pair_sim_jaccard_resource**](docs/PairsimApi.md#get_pair_sim_jaccard_resource) | **GET** /pair/sim/jaccard/{id1}/{id2} | Get pairwise similarity
*RelationusageApi* | [**get_relation_usage_between_resource**](docs/RelationusageApi.md#get_relation_usage_between_resource) | **GET** /relation/usage/between/{subject_category}/{object_category} | All relations used plus count of associations
*RelationusageApi* | [**get_relation_usage_pivot_label_resource**](docs/RelationusageApi.md#get_relation_usage_pivot_label_resource) | **GET** /relation/usage/pivot/label | Relation usage count for all subj x obj category combinations, showing label
*RelationusageApi* | [**get_relation_usage_pivot_resource**](docs/RelationusageApi.md#get_relation_usage_pivot_resource) | **GET** /relation/usage/pivot | Relation usage count for all subj x obj category combinations
*RelationusageApi* | [**get_relation_usage_resource**](docs/RelationusageApi.md#get_relation_usage_resource) | **GET** /relation/usage/ | All relations used plus count of associations
*SearchApi* | [**get_autocomplete**](docs/SearchApi.md#get_autocomplete) | **GET** /search/entity/autocomplete/{term} | Returns list of matching concepts or entities using lexical search
*SearchApi* | [**get_search_entities**](docs/SearchApi.md#get_search_entities) | **GET** /search/entity/{term} | Returns list of matching concepts or entities using lexical search
*SearchApi* | [**get_search_hpo_entities**](docs/SearchApi.md#get_search_hpo_entities) | **GET** /search/entity/hpo-pl/{term} | Returns list of matching concepts or entities using lexical search
*SimApi* | [**get_annotation_score**](docs/SimApi.md#get_annotation_score) | **GET** /sim/score | Get annotation score
*SimApi* | [**get_sim_compare**](docs/SimApi.md#get_sim_compare) | **GET** /sim/compare | Compare a reference profile vs one profiles
*SimApi* | [**get_sim_search**](docs/SimApi.md#get_sim_search) | **GET** /sim/search | Search for phenotypically similar diseases or model genes
*SimApi* | [**post_annotation_score**](docs/SimApi.md#post_annotation_score) | **POST** /sim/score | Get annotation score
*SimApi* | [**post_sim_compare**](docs/SimApi.md#post_sim_compare) | **POST** /sim/compare | Compare a reference profile vs one or more profiles
*VariationsetApi* | [**delete_variant_set_item**](docs/VariationsetApi.md#delete_variant_set_item) | **DELETE** /variation/set/{id} | Deletes variant set
*VariationsetApi* | [**get_variant_analyze**](docs/VariationsetApi.md#get_variant_analyze) | **GET** /variation/set/analyze/{id} | Returns list of matches
*VariationsetApi* | [**get_variant_set_item**](docs/VariationsetApi.md#get_variant_set_item) | **GET** /variation/set/{id} | Returns a variant set
*VariationsetApi* | [**get_variant_sets_archive_collection**](docs/VariationsetApi.md#get_variant_sets_archive_collection) | **GET** /variation/set/archive/{year}/{month}/{day} | Returns list of variant sets from a specified time period
*VariationsetApi* | [**get_variant_sets_collection**](docs/VariationsetApi.md#get_variant_sets_collection) | **GET** /variation/set/ | Returns list of variant sets
*VariationsetApi* | [**post_variant_sets_collection**](docs/VariationsetApi.md#post_variant_sets_collection) | **POST** /variation/set/ | Creates a new variant set
*VariationsetApi* | [**put_variant_set_item**](docs/VariationsetApi.md#put_variant_set_item) | **PUT** /variation/set/{id} | Updates a variant set

## Documentation For Models

 - [APageOfResults](docs/APageOfResults.md)
 - [AbstractPropertyValue](docs/AbstractPropertyValue.md)
 - [AllOfAnnotationExtensionFiller](docs/AllOfAnnotationExtensionFiller.md)
 - [AllOfAssociationEvidenceGraph](docs/AllOfAssociationEvidenceGraph.md)
 - [AllOfAssociationObject](docs/AllOfAssociationObject.md)
 - [AllOfAssociationObjectExtensionsItems](docs/AllOfAssociationObjectExtensionsItems.md)
 - [AllOfAssociationRelation](docs/AllOfAssociationRelation.md)
 - [AllOfAssociationSubject](docs/AllOfAssociationSubject.md)
 - [AllOfAssociationSubjectExtensionsItems](docs/AllOfAssociationSubjectExtensionsItems.md)
 - [AllOfPairwiseMatchLcs](docs/AllOfPairwiseMatchLcs.md)
 - [AllOfPairwiseMatchMatch](docs/AllOfPairwiseMatchMatch.md)
 - [AllOfPairwiseMatchReference](docs/AllOfPairwiseMatchReference.md)
 - [AllOfSimQueryIdsItems](docs/AllOfSimQueryIdsItems.md)
 - [AllOfSimQueryNegatedIdsItems](docs/AllOfSimQueryNegatedIdsItems.md)
 - [AllOfSimQueryReference](docs/AllOfSimQueryReference.md)
 - [AllOfSimResultMatchesItems](docs/AllOfSimResultMatchesItems.md)
 - [AnnotationExtension](docs/AnnotationExtension.md)
 - [Association](docs/Association.md)
 - [AssociationResults](docs/AssociationResults.md)
 - [AutocompleteResult](docs/AutocompleteResult.md)
 - [AutocompleteResults](docs/AutocompleteResults.md)
 - [BioObject](docs/BioObject.md)
 - [BioObjectCore](docs/BioObjectCore.md)
 - [CompactAssociationSet](docs/CompactAssociationSet.md)
 - [CompareInput](docs/CompareInput.md)
 - [D2PAssociation](docs/D2PAssociation.md)
 - [D2PAssociationResults](docs/D2PAssociationResults.md)
 - [Edge](docs/Edge.md)
 - [EntityAnnotationResult](docs/EntityAnnotationResult.md)
 - [EntityReference](docs/EntityReference.md)
 - [Feature](docs/Feature.md)
 - [Graph](docs/Graph.md)
 - [IcNode](docs/IcNode.md)
 - [LayAutocomplete](docs/LayAutocomplete.md)
 - [LayResults](docs/LayResults.md)
 - [Mme](docs/Mme.md)
 - [NamedObject](docs/NamedObject.md)
 - [NamedObjectCore](docs/NamedObjectCore.md)
 - [Node](docs/Node.md)
 - [PageOfVariantSets](docs/PageOfVariantSets.md)
 - [PairwiseMatch](docs/PairwiseMatch.md)
 - [Relation](docs/Relation.md)
 - [RelationRef](docs/RelationRef.md)
 - [SearchResult](docs/SearchResult.md)
 - [Seq](docs/Seq.md)
 - [SequenceFeature](docs/SequenceFeature.md)
 - [SequenceLocation](docs/SequenceLocation.md)
 - [SequencePosition](docs/SequencePosition.md)
 - [SimMatch](docs/SimMatch.md)
 - [SimMetadata](docs/SimMetadata.md)
 - [SimQuery](docs/SimQuery.md)
 - [SimResult](docs/SimResult.md)
 - [Span](docs/Span.md)
 - [SufficiencyOutput](docs/SufficiencyOutput.md)
 - [SufficiencyPostInput](docs/SufficiencyPostInput.md)
 - [SynonymPropertyValue](docs/SynonymPropertyValue.md)
 - [Taxon](docs/Taxon.md)
 - [Token](docs/Token.md)
 - [TypedNode](docs/TypedNode.md)
 - [VariantSet](docs/VariantSet.md)

## Documentation For Authorization

 All endpoints do not require authorization.


## Author


